kernel/src/mm/physical_memory_manager.rs

```diff
@@ -4,12 +4,33 @@ use crate::mm::PAddr;
 use crate::Architecture;
 use core::mem;
 
+/// A range similar to core::ops::Range but that is copyable.
+/// The range is half-open, inclusive below, excluse above, ie. [start; end[
+#[derive(Debug, Copy, Clone, PartialEq)]
+struct AddressRange<T> {
```

Comment:
```suggestion
struct AddressRange<T: Copy> {
```
to make sure that the range is copyable. This will ensure that people aren't able to create an `AdressRange` by hand but not use any of the methods on it

Reply:
What ?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2f1c26bb449025501e76cc9df151a774

If the struct's generic and the impl's generic do not have the same trait bounds, especially when the struct's fields are public, you can create interesting errors. But this is also very helpful when you want to specialize the implementation based on trait bounds